backend/src/controllers/searchController.js

```diff
@@ -104,8 +100,40 @@ function addProductFeatureFilter(filter, requiredFeatureIds) {
   return filter;
 }
 
+/**
+ * Generates base productFeature filter for filtering through product id's in products
+ * @param featureId
+ * @returns {{productFeature: {$eq: *}}}
+ */
 function generateBaseProductFeatureFilter(featureId) {
   return { productFeature: { $eq: featureId } };
 }
 
+/**
+ *
+ * @param productFilter - what to filter the products on
+ * @param productFeatureFilter - what to filter the features in the products on (not the entire feature documents;
+ * the nested feature metadata
+ * @param featureFilter - what to filter the entire feature documents on
+ * @param targetPosition - the "target" position of the output features. Features will be returned in
+ * order of how close they are to this target position
+ * @returns {*|Aggregate}
+ */
+function aggregateProductFeaturesFromProducts(productFilter, productFeatureFilter, featureFilter, targetPosition) {
+  return Product.aggregate()
+    .match(productFilter)
+    .unwind('features')
+    .replaceRoot('features')
+    .match(productFeatureFilter)
+    .addFields({ [DIFF_FIELD]: { $abs: { $subtract: ['$name.medianIndex', targetPosition] } } }) // calculate the diff from the target position
+    .group({ _id: '$productFeature', [DIFF_FIELD]: { $min: '$' + DIFF_FIELD } }) // group matching product features. keep the one with the lowest delta
+    .lookup({ from: 'productfeatures', localField: '_id', foreignField: '_id', as: 'productFeature' })
+    .unwind('productFeature')
+    .replaceRoot({ $mergeObjects: ['$productFeature', '$$ROOT'] }) // merge the root with the productFeature object because we want to keep the DIFF_FIELD (and maybe other calculated fields in the future)
+    .project({ productFeature: 0 }) // it's attributes go merged in, remove it
+    .match(featureFilter)
+    .sort({ [DIFF_FIELD]: 1, count: -1 }) // sort by position diff
```

Comment:
Recalculate local frequency/count and sort by that instead of the global frequency field